core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,423 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+      } else {
+         // Else the user will have a warning and the URL in the terminal
+         Warning("Help", "The DISPLAY is not set!");
+         std::cout << url.Data() << std::endl;
+      }
+   }
+}
+namespace {
+static TString URLGenerator(TString ClassName)
+{
+   // Generates a URL address for the class (classname).
+   // This is the URL to the online reference guide, generated by Doxygen.
+   TString URLBegining("https://root.cern.ch/doc/master/class");
+   TString URLEnd(".html");
+   URLBegining.Append(ClassName);
+   URLBegining.Append(URLEnd);
+   return URLBegining.Data();
+}
+}
+
+namespace {
+static TString URLforNameSpace(TString NameSpace)
+{
+   // Generates a URL address for the namespace (NameSpace).
+   // This is the URL to the online reference guide, generated by Doxygen.
+   TString URLbegining("https://root.cern/doc/master/namespace");
+   TString URLend(".html");
+   URLbegining.Append(NameSpace.Data());
+   URLbegining.Append(URLend);
+   return URLbegining.Data();
+}
+}
+
+namespace {
+static TString DoxygenSyntax(TString ClassName, TString Method)
+{
+   // The function returns a TString with the arguments of a member function (Method)
+   // of a class (ClassName), but modified with respect to the Doxygen syntax - spacing
+   // around special sumbols and adding the missing sumbols ("std::", ",...").
+   // "DoxygenSyntax" works for the functions defined inside namespaces too.
+   TString MethodArguments(((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->GetSignature());
+   // With "GetSignature" we get the arguments of "Method" and put them in a TString.
+   // Then the TString "MethodArguments" is modified.
+   MethodArguments.ReplaceAll(" = ", "=");
+   // maybe add also " =" and "= ";
+   MethodArguments.ReplaceAll("* ", " *");
+   // is this always? below
+   MethodArguments.ReplaceAll("*=", " *=");
+   MethodArguments.ReplaceAll("*)", " *)");
+   MethodArguments.ReplaceAll("*,", " *,");
+   MethodArguments.ReplaceAll("*& ", " *&");
+   MethodArguments.ReplaceAll("& ", " &");
+   MethodArguments.ReplaceAll("ostream", "std::ostream");
+   MethodArguments.ReplaceAll("istream", "std::istream");
+   MethodArguments.ReplaceAll("*msgfmt", "*msgfmt,...");
+   TString Buff ("\\b");
+   Buff.Append(ClassName);
+   Buff.Append("::\\b");
+   TPRegexp ArgFix(Buff.Data());
+   ArgFix.Substitute(MethodArguments, "");
+   return MethodArguments;
+}
+}
+
+namespace {
+static TString 
+InlineFuncCheck(TString ClassName, TString Method)
+{
+   // The function checks if a member function (Method) of a class (ClassName) is
+   // defined as inline. If so, it also checks if it is virtual.
+   // Then the return type of "Method" is changed for the need of Doxygen and with
+   // respect to the function definition. The corrected "ReturnType" is returned.
+
+   Int_t InlineCheck = (((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->ExtraProperty() & kIsInlined);
```

Comment:
Please split this up:
```
TClass *cl = TClass::GetClass(ClassName);
if (!cl) {
  Error(...);
  return "";
}
TMethod *meth = cl->GetListOfMethods()->FindObject(Method);
if (!meth) {
...
}
```

Reply:
Maybe even better: `TString AdjustReturnTypeForDoxygen(const TString &className, TClass *cl, const TString &methodName, TMethod *method)`. Like that we don't need to do `TClass::GetClass()` *again*.